### PR TITLE
The function "getAttachedData" has been removed

### PR DESCRIPTION
**Pull request by annando**
Needs https://git.friendi.ca/friendica/friendica-addons/pulls/1359 to be merged.

The function `BBCode::getAttachedData` was totally outdated and did a lot of unnecessary stuff.

### CODE REVIEW

File: src/Content/Item.php [hunk 1]
```diff
@@ -223,15 +223,19 @@ public static function replaceTag(string &$body, int $profile_uid, string $tag,
 
 				// select someone by nick in the current network
 				if (!DBA::isResult($contact) && ($network != '')) {
-					$condition = ["`nick` = ? AND `network` = ? AND `uid` = ?",
-						$name, $network, $profile_uid];
+					$condition = [
+						"`nick` = ? AND `network` = ? AND `uid` = ?",
+						$name, $network, $profile_uid
+					];
```
Comment (MrPetovan) [inline src/Content/Item.php]: Can't this be simplified to:
```suggestion
					$condition = [
						'nick'    => $name,
						'network' => $network,
						'uid'     => $profile_uid,
					];
```
?

File: src/Content/Item.php [hunk 2]
```diff
@@ -223,15 +223,19 @@ public static function replaceTag(string &$body, int $profile_uid, string $tag,
...
 				}
 
 				// select someone by attag in the current network
 				if (!DBA::isResult($contact) && ($network != '')) {
-					$condition = ["`attag` = ? AND `network` = ? AND `uid` = ?",
-						$name, $network, $profile_uid];
+					$condition = [
+						"`attag` = ? AND `network` = ? AND `uid` = ?",
+						$name, $network, $profile_uid
+					];
```
Comment (MrPetovan) [inline src/Content/Item.php]:
```suggestion
					$condition = [
						'attag'   => $name,
						'network' => $network,
						'uid'     => $profile_uid,
					];
```

File: src/Content/Item.php [hunk 3]
```diff
@@ -333,7 +339,8 @@ public function localize(array &$item)
 					default:
 						if ($obj['resource-id']) {
 							$post_type = $this->l10n->t('photo');
-							$m=[]; preg_match("/\[url=([^]]*)\]/", $obj['body'], $m);
+							$m = [];
+							preg_match("/\[url=([^]]*)\]/", $obj['body'], $m);
 							$rr['plink'] = $m[1];
```
Comment (MrPetovan) [inline src/Content/Item.php]: Initialization is superfluous
```suggestion
							preg_match("/\[url=([^]]*)\]/", $obj['body'], $matches);
							$rr['plink'] = $matches[1];
```

File: src/Content/Text/BBCode.php [hunk 4]
```diff
@@ -1113,7 +871,7 @@ function ($match) use ($simplehtml, $uriid) {
 				}
 
 				$img_str = '<img src="' .
-				self::proxyUrl($match[2], $simplehtml, $uriid) . '"';
+					self::proxyUrl($match[2], $simplehtml, $uriid) . '"';
```
Comment (MrPetovan) [inline src/Content/Text/BBCode.php]: New line is superfluous here, it should be combined with line 873.

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Content/Text/BBCode.php (hunk 4 above)
Comment (annando) [reply src/Content/Text/BBCode.php]: Done

File: src/Content/Item.php (hunk 3 above)
Comment (annando) [reply src/Content/Item.php]: Done

File: src/Content/Item.php (hunk 2 above)
Comment (annando) [reply src/Content/Item.php]: Done

File: src/Content/Item.php (hunk 1 above)
Comment (annando) [reply src/Content/Item.php]: Done

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**